services/EventAPIService.js

```diff
@@ -0,0 +1,71 @@
+const axios = require("axios");
+
+// A service class for all REST API operations
+class EventAPIService {
+  #axiosInstance;
+  #tokenServiceInstance;
+  constructor(tokenService) {
+    // Set Axios baseURL to 100ms API BaseURI
+    this.#axiosInstance = axios.create({
+      baseURL: "https://api.100ms.live/v2",
+      timeout: 3 * 60000,
+    });
+    this.#tokenServiceInstance = tokenService;
+    this.#configureAxios();
+  }
+
+  // Add Axios interceptors to process all requests and responses
+  #configureAxios() {
+    this.#axiosInstance.interceptors.request.use(
+      (config) => {
+        // Add Authorization on every request made using the Management token
+        config.headers = {
+          Authorization: `Bearer ${this.#tokenServiceInstance.getManagementToken()}`,
+          Accept: "application/json",
+          "Content-Type": "application/json",
+        };
+        return config;
+      },
+      (error) => Promise.reject(error)
+    );
+    this.#axiosInstance.interceptors.response.use(
+      (response) => {
+        return response;
+      },
+      (error) => {
+        logger.error("Error in making API call", { response: error.response?.data });
+        const originalRequest = error.config;
+        if ((error.response?.status === 403 || error.response?.status === 401) && !originalRequest._retry) {
+          logger.info("Retrying request with refreshed token");
```

Comment:
NIT: to improve the code readability, we can move these conditions inside if stmt to a boolean function.
As the no of condition grows, it becomes harder to read.

Reply:
Fixed it. Thank you!